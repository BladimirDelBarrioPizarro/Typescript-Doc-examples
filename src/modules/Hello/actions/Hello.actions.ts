import {
    Message,
    SEND_MESSAGE,
    HelloActionTypes
} from '../types/Hello.types'

export const helloMessageAction = (newMessage:Message):HelloActionTypes => {
    return{
        type:SEND_MESSAGE,
        payload:newMessage
    }
}
