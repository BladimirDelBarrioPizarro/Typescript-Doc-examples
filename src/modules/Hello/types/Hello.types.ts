export const SEND_MESSAGE = 'SEND_MESSAGE'

export interface Message {
    type: string;
    enthusiasmLevel: number
  }

export interface IStateHello {
    data:Message[]
}

export interface SendMessageAction{
    type:typeof SEND_MESSAGE,
    payload: Message
}

export type HelloActionTypes = SendMessageAction
