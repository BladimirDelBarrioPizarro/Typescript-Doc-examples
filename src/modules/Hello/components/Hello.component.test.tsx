import * as React from "react";
import * as Enzyme from "enzyme";
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import {Provider} from 'react-redux'
import store from '../../../store/store'

import Hello from "./Hello.component";
import * as Helloactions from '../actions/Hello.actions'
import * as Hellotypes from '../types/Hello.types'
import { render } from "@testing-library/react";

configure({ adapter: new Adapter() });






describe('renders Hello component ', () => {
    it("renders the correct text with an explicit enthusiasm of 1", () => {
        
        const wrapper = Enzyme.shallow(
           <Provider store={store}>
               <Hello name="Bladi" enthusiasmLevel={1} />
           </Provider> 
            
        ).dive()
        expect(wrapper).toMatchSnapshot()
      });

      it("throws when the enthusiasm level is 0", () => {
        expect(() => {
          Enzyme.shallow(<Hello name="Bladi" enthusiasmLevel={0} />);
        }).toThrow();
      });
      
      it("throws when the enthusiasm level is negative", () => {
        expect(() => {
          Enzyme.shallow(<Hello name="Bladi" enthusiasmLevel={-1} />);
        }).toThrow();
      }); 
})


/*
it("renders the correct text with an explicit enthusiasm level of 5", () => {
  const hello = Enzyme.shallow(<Hello name="Bladi" enthusiasmLevel={5} />);
  expect(hello.find(".greeting").text()).toEqual("Hello Bladi!!!!!");
});

it("throws when the enthusiasm level is 0", () => {
  expect(() => {
    Enzyme.shallow(<Hello name="Bladi" enthusiasmLevel={0} />);
  }).toThrow();
});

it("throws when the enthusiasm level is negative", () => {
  expect(() => {
    Enzyme.shallow(<Hello name="Bladi" enthusiasmLevel={-1} />);
  }).toThrow();
});
 */

describe('Hello.actions', () => {
    it('should create an action SEND_MESSAGE', () => {
      const data:Hellotypes.Message = {
          message:'Bladi',
          enthusiasmLevel:1
      }
      const expectedAction = {
        type: Hellotypes.SEND_MESSAGE,
        payload:data
      }
      expect(Helloactions.helloMessageAction(data)).toEqual(expectedAction)
    })
  })