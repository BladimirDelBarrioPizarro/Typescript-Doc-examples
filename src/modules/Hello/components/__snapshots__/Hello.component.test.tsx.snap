// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Hello component  renders the correct text with an explicit enthusiasm of 1 1`] = `ShallowWrapper {}`;
